<?xml version="1.0" encoding="UTF-8" ?>

<launch>
    <arg name="veh" default="jackal"/>
    <arg name="config_file" default="grid_map_info"/>
    <arg name="map_frame" default="map"/>
    <arg name="robot_frame" default="base_link"/>
    <arg name="use_odom" default="False"/>
    <arg name="use_tf" default="True"/>
    <arg name="sim" default="False"/>

    <group ns="$(arg veh)">
    <node pkg="astar" type="lidarpc2grid" name="lidarpc2grid" output="screen">
        <remap from="points" to="/velodyne1/velodyne_points" />
        <rosparam command="load" file="$(find astar)/config/$(arg config_file).yaml" />
    </node>
    <node pkg="astar" type="path_plan" name="plan" output="screen">
        <remap from="map" to="pc_map" />
    </node>

    <node pkg="astar" type="tracking_guid_astar.py" name="track" output="screen">
      <param name="map_frame" value="$(arg map_frame)" />
      <param name="lookahead" value="0.7" />
      <remap from="joy_teleop/joy" to="/bluetooth_teleop/joy" />
    </node>

    <!-- <node pkg="astar" type="odom_tf.py" name="odom_tf" output="screen">
        <remap from="husky_velocity_controller/odom" to="/jackal_velocity_controller/odom" />
    </node> -->

    <node pkg="control" type="pid_control_ncsist.py" name="pid_control" output="screen">
        <rosparam param="use_odom" subst_value="True">False</rosparam>
        <rosparam param="use_tf" subst_value="True">True</rosparam>
        <param name="map_frame" value="$(arg map_frame)"/>
        <param name="robot_frame" value="$(arg robot_frame)"/>
        <param name="cmd_ratio" value="0.65"/>
        <remap from="cmd_vel" to="/cmd_vel"/>
        <remap from="pose" to="/jackal_velocity_controller/odom"/>
        <remap from="joy_teleop/joy" to="/bluetooth_teleop/joy" />
</node>
</group>
</launch>
